### PR TITLE
feat(search bar): add search bar field and functionality

### LINKED ISSUE #5: Add a search bar on top
Below `Conferences, but in your home.` it would be nice to have a search bar that allows free text search on the name and description of the event.

Something really minimal and easy to use

### PR DESCRIPTION
**Pull request by namadnuno**
Closes #5

### CODE REVIEW

File: src/pages/index.js [hunk 1]
```diff
@@ -7,41 +7,75 @@ import Container from '../components/Container'
...
+            : false
+          let inTitle = conference.name
+            ? conference.name.toLowerCase().indexOf(query.toLowerCase()) >= 0
+            : false
+          return inDescription || inTopic || inTitle
+        })
+      )
+      setLoading(false)
+    } catch (e) {
+      console.log('error', e)
```
Comment (luisFilipePT) [inline src/pages/index.js]: Please remove logs (only for development)

**Review by luisFilipePT — COMMENTED** [review]
It looks really cool and I think this is want we want 😄 thanks but:

- We already have in the `state` all conferences (at least in the current version) so there is no need to `fetch` again,, can you keep the logic but work only with the state when searching? To avoid unneeded network request.

- Did you test it on mobile?

- Maybe add a button to trigger the search or throttle for a time interval and search without pressing enter (I prefer the second one but anyone will do)

- Can you add a reset icon to clear the search?

Thanks 🙏

File: src/pages/index.js (hunk 1 above)
Comment (namadnuno) [reply src/pages/index.js]: It was already there, but i can remove it

**Review by coderbyheart — APPROVED** [review]
LGTM

I would have used debounce and autosubmit after 250ms, which is more what I would expect from an in-site search.

**Comment by luisFilipePT** [comment]
@coderbyheart @namadnuno  is gonna open a new PR with those enhancements 😄 .

**Comment by luisFilipePT** [comment]
@coderbyheart next time in this case I will `request changes` to block the `merge` 😄

**Comment by coderbyheart** [comment]
Yes, or mark it as "Draft"!

I like to move fast, ideally every commit should go to trunk: https://trunkbaseddevelopment.com/

**Pull request merged**